src/katgpucbf/xbgpu/engine.py

```diff
@@ -496,14 +510,18 @@ async def _receiver_loop(self) -> None:
 
         The above steps are performed in a loop until there are no more chunks to assembled.
         """
-        async for chunk in recv.recv_chunks(self.receiver_stream):
-            timestamp = chunk.chunk_id * self.rx_heap_timestamp_step * self.heaps_per_fengine_per_chunk
-
+        async for chunk in recv.recv_chunks(
+            self.receiver_stream,
+            self._src_layout,
+            self.sensors,
+            self.time_converter,
+        ):
             # Get a free rx_item that will contain the GPU buffer to which the
             # received chunk will be transferred.
             item = await self._rx_free_item_queue.get()
-            item.timestamp += timestamp
             item.chunk = chunk
+            # Need a seperate attribute as the chunk gets reset
+            item.timestamp += chunk.timestamp
```

Comment:
I'm a bit confused about why it is `+=` rather than `=` (and I realise that's what the original code did). Adding (absolute) timestamps makes no sense to me.